crux-agent/app/core/agents/evaluator.py

```
@@ -137,8 +137,15 @@ async def run(self, context: AgentContext) -> AgentResult:
-            # Log the raw evaluation prompt for debugging
-            logger.info(f"Raw evaluation_prompt: {evaluation_prompt}")
+            # Log the evaluation prompt compacted (collapse whitespace) at INFO for readability
+            compact_prompt = re.sub(r"\s+", " ", evaluation_prompt).strip()
+            compact_prompt_preview = (compact_prompt[:2000] + "...") if len(compact_prompt) > 2000 else compact_prompt
+            logger.info(
+                f"Evaluation prompt (orig_len={len(evaluation_prompt)}, compact_len={len(compact_prompt)}): {compact_prompt_preview}"
+            )
+            # Also keep a truncated raw at DEBUG if needed
+            raw_prompt_preview = (evaluation_prompt[:2000] + "...") if len(evaluation_prompt) > 2000 else evaluation_prompt
+            logger.debug(f"Evaluation prompt RAW (len={len(evaluation_prompt)}): {raw_prompt_preview}")
             
```

Ruff (0.12.2)

Blank line contains whitespace

Remove whitespace from blank line

(W293)